tests/recipes/test_lora_dpo_single_device.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```

Comment:
You're a hero

Reply:
I was just afraid someone would ask me to run the recipe manually, so this felt like less effort.